fastjet/internal/delaunay/point.go

```diff
@@ -63,6 +63,62 @@ func (p *Point) Equals(v *Point) bool {
 	return p == v || (p.x == v.x && p.y == v.y)
 }
 
+// SecondNearestNeighbor looks at all adjacent points of p and returns the second nearest one
+// and the distance to that point.
+func (p *Point) SecondNearestNeighbor() (*Point, float64) {
+	var nearest, secondNearest *Point
+	min, secondMin := math.Inf(1), math.Inf(1)
+	if p.dist2 == 0 {
+		// p has a duplicate
+		nearest = p.nearest
+		min = 0
+	}
+	for _, t := range p.adjacentTriangles {
+		var p2, p3 *Point
+		// find the point in t that is not p
+		switch {
+		case p.Equals(t.A):
+			p2 = t.B
+			p3 = t.C
+		case p.Equals(t.B):
+			p2 = t.A
+			p3 = t.C
+		case p.Equals(t.C):
+			p2 = t.A
+			p3 = t.B
+		default:
+			panic(fmt.Errorf("delaunay: point %v not found in %v", p, t))
+		}
+		dist := p.distance(p2)
+		if dist < min {
```

Comment:
I would use a `switch` instead of that `if-else-if`:

```go
switch {
case dist < min:
case dist < secondMin:
}
```

Reply:
done